### ISSUE TITLE
ZoneIngress fails to sync to zone due to limitation on kubernetes label length

### ISSUE THREAD

**Issue by ttreptow** [kumahq/kuma#9168]
### What happened?

When syncing a ZoneIngress with a long name from the global control plane, the zone control plane complains that metadata.labels cannot be longer than 63 characters:

> 2024-02-07T17:43:00.180Z	INFO	kds-delta-zone	error during callback received, sending NACK	{"kds-version": "v2", "err": "failed to create k8s resource: ZoneIngress.kuma.io \"domain-discovery-onprem.domain-discovery-aws.kuma-ingress-5bc8f5f985-96hjz.kuma.default-2wwx5vw4c9fwc8zx\" is invalid: metadata.labels: Invalid value: \"domain-discovery-onprem.domain-discovery-aws.kuma-ingress-5bc8f5f985-96hjz.kuma.default-2wwx5vw4c9fwc8zx\": must be no more than 63 characters", "errVerbose": "ZoneIngress.kuma.io \"domain-discovery-onprem.domain-discovery-aws.kuma-ingress-5bc8f5f985-96hjz.kuma.default-2wwx5vw4c9fwc8zx\" is invalid: metadata.labels: Invalid value: \"domain-discovery-onprem.domain-discovery-aws.kuma-ingress-5bc8f5f985-96hjz.kuma.default-2wwx5vw4c9fwc8zx\": must be no more than 63 characters\nfailed to create k8s resource\ngithub.com/kumahq/kuma/pkg/plugins/resources/k8s.(*KubernetesStore).Create\n\tgithub.com/kumahq/kuma/pkg/plugins/resources/k8s/store.go:80\ngithub.com/kumahq/kuma/pkg/core/resources/store.(*paginationStore).Create\n\tgithub.com/kumahq/kuma/pkg/core/resources/store/pagination_store.go:30\ngithub.com/kumahq/kuma/pkg/metrics/store.(*MeteredStore).Create\n\tgithub.com/kumahq/kuma/pkg/metrics/store/store.go:44\ngithub.com/kumahq/kuma/pkg/core/resources/store.(*customizableResourceStore).Create\n\tgithub.com/kumahq/kuma/pkg/core/resources/store/customizable_store.go:44\ngithub.com/kumahq/kuma/pkg/kds/v2/store.(*syncResourceStore).Sync.func2\n\tgithub.com/kumahq/kuma/pkg/kds/v2/store/sync.go:230\ngithub.com/kumahq/kuma/pkg/core/resources/store.InTx\n\tgithub.com/kumahq/kuma/pkg/core/resources/store/transactions.go:46\ngithub.com/kumahq/kuma/pkg/kds/v2/store.(*syncResourceStore).Sync\n\tgithub.com/kumahq/kuma/pkg/kds/v2/store/sync.go:205\ngithub.com/kumahq/kuma/pkg/kds/zone.Setup.func2.ZoneSyncCallback.func3\n\tgithub.com/kumahq/kuma/pkg/kds/v2/store/sync.go:305\ngithub.com/kumahq/kuma/pkg/kds/v2/client.(*kdsSyncClient).Receive\n\tgithub.com/kumahq/kuma/pkg/kds/v2/client/kds_client.go:90\ngithub.com/kumahq/kuma/pkg/kds/zone.Setup.func2.1\n\tgithub.com/kumahq/kuma/pkg/kds/zone/components.go:134\nruntime.goexit\n\truntime/asm_amd64.s:1650"}

In this multi-zone setup there is a global control plane and two zones, `domain-discovery-aws` and `domain-discovery-onprem` 

## Info
Kuma version: 2.6.0 (dp, cp, zone ingress all have the same version)
kuberntes: 1.28.4
Kuma was previously running 2.4.3 and was upgraded to 2.6.0

**Comment by lobkovilya** [comment]
what was the original ZoneIngress name in `domain-discovery-onprem` zone (before it was synced to Global)?

**Comment by ttreptow** [comment]
I re-upgraded to 2.6.0 to check and it the ZoneIngress name is just the pod name `kuma-ingress-6579d94fdd-ksd6l` (the pod has been restarted so it's different than in the log above). I'm not seeing the above error this time; however it is stuck on the same issue as #9006. In my case it doesn't clear after a time though. The last time I upgraded I also ran into that issue and after clearing it (somehow, don't remember how) I ran into the above issue. 
I will continue trying to solve the duplicate listener issue and when that is cleared I'll check if the label issue shows up again.
